custom/icds_reports/data_pull/exporter.py

```diff
@@ -30,9 +31,14 @@ def __init__(self, slug_or_file, db_alias, month, location_id):
     @prevent_parallel_execution(DATA_PULL_CACHE_KEY)
     def export(self):
         zip_file_name = "%s-DataPull.zip" % self.data_pull_obj.name
-        with zipfile.ZipFile(zip_file_name, mode='a') as z:
-            for filename, string_buffer in self.data_pull_obj.run().items():
-                z.writestr(filename, string_buffer.getvalue())
+        if self.data_pull_obj.file_format_required() == 'xlsx':
+            with zipfile.ZipFile(zip_file_name, "w") as zip_content:
+                for filename, workbook in self.data_pull_obj.run().items():
+                    zip_content.writestr(filename, get_file_content_from_workbook(workbook))
+        else:
+            with zipfile.ZipFile(zip_file_name, mode='a') as z:
+                for filename, string_buffer in self.data_pull_obj.run().items():
+                    z.writestr(filename, string_buffer.getvalue())
```

Comment:
I think "w" is the correct mode and "a" was an oversight earlier, so we should keep "w" only.

Why not keep the same approach of returning filestream to avoid handling this here?

Reply:
could not find a way to return workbook as a stringIO so added a method to reduce the repeated code